pkg/controller/database/claim.go

```diff
@@ -199,21 +201,17 @@ func ConfigureMyCloudsqlInstance(_ context.Context, cm resource.Claim, cs resour
 	return nil
 }
 
-func translateVersion(version, versionPrefix string) string {
-	if version == "" {
-		return ""
+func translateVersion(version, versionPrefix string) *string {
+	r := ""
+	if version != "" {
+		r = fmt.Sprintf("%s_%s", versionPrefix, strings.Replace(version, ".", "_", -1))
 	}
-	return fmt.Sprintf("%s_%s", versionPrefix, strings.Replace(version, ".", "_", -1))
+	return &r
```

Comment:
Should this return `nil` rather than a pointer to empty string, given `DatabaseVersion` is an optional field?